### ISSUE TITLE
Slow access to JavaKeyStore

### ISSUE THREAD

**Issue by lucabelluccini** [elastic/logstash#10126]
- Version: 6.4.1
- Config File:
```
config.reload.automatic: true 
config.reload.interval: 3s 
```
- Steps to Reproduce:
  - Make use of the JavaKeyStore for some secure setting
  - Reload a pipeline and wait for the autoreload

The Logstash instance consumes an average of 3s reading the JavaKeyStore. This is done several times during a restart.
```
[2018-11-06T09:24:08,226][DEBUG][org.logstash.secret.store.SecretStoreFactory] Attempting to exists or secret store with implementation: org.logstash.secret.store.backend.JavaKeyStore
[2018-11-06T09:24:11,596][DEBUG][org.logstash.secret.store.SecretStoreFactory] Attempting to load or secret store with implementation: org.logstash.secret.store.backend.JavaKeyStore
```

@guyboertje mentioned the fact there is brute-force-attack defence in the Java KeyStore implementations that slow down repeated access.

**Comment by yaauie** [comment]
This is a substantial problem in large pipelines that use substitution variables extensively, even if the user is providing the variables via the environment and not via the keystore; for each ~100 replacement variables in the pipeline configurations, this "defence" adds _five minutes_ of overhead to each pipeline reload (including the initial pipeline load at startup).

**Comment by yaauie** [comment]
A fix for this will land in 8.0, 7.8, and the as-yet-unscheduled 6.8.9 via https://github.com/elastic/logstash/pull/10794, and in 7.7.1 via https://github.com/elastic/logstash/pull/11772
